client/my-sites/plugins/plugin-upload/index.jsx

```diff
@@ -96,17 +106,19 @@ export default connect(
 		const siteId = getSelectedSiteId( state );
 		const error = getPluginUploadError( state, siteId );
 		const progress = getPluginUploadProgress( state, siteId );
+		const isJetpack = isJetpackSite( state, siteId );
 		return {
 			siteId,
 			siteSlug: getSelectedSiteSlug( state ),
-			isJetpack: isJetpackSite( state, siteId ),
+			isJetpack,
 			inProgress: isPluginUploadInProgress( state, siteId ),
 			complete: isPluginUploadComplete( state, siteId ),
 			failed: !! error,
 			pluginId: getUploadedPluginId( state, siteId ),
 			error,
 			progress,
 			installing: progress === 100,
+			upgradeJetpack: isJetpack && ! isJetpackMinimumVersion( state, siteId, 5.1 ),
```

Comment:
Let's make sure we pass the version as a string to the `isJetpackMinimumVersion` selector.